src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -779,6 +779,31 @@ def projection_in_str(self) -> str:
             )
         return self._projection_in_str
 
+    def with_subqueries(self, subquery_plans: List[SnowflakePlan]) -> "SelectStatement":
+        """Update pre-actions, post-actions and schema to capture necessary subquery_plans
+        encountered during plan resolution."""
+        for plan in subquery_plans:
+            for query in plan.queries[:-1]:
+                if self.pre_actions is None:
+                    self.pre_actions = []  # pragma: no cover
+                if query not in self.pre_actions:
+                    self.pre_actions.append(query)
+            for query in plan.post_actions:
+                if self.post_actions is None:
+                    self.post_actions = []  # pragma: no cover
+                if query not in self.post_actions:
+                    self.post_actions.append(query)
+
+            if (self._schema_query is not None) and (plan.schema_query is not None):
```

Comment:
should call self.schema_query for this, instead of calling the cached one, otherwise if the schema_query cache is None, when calling schema_query, we will still see the same problem.

Reply:
This is how we create `schema_query` for different `Selectable` classes

1. `SelectableEntity`: `SELECT * FROM ENTITY`
2. `SelectSQL`: `sql_query`
3. `SetStatement`: build a schema query using `attributes`
4. `SelectSnowflakePlan`: `self.snowflake_plan.schema_query`
5. `SelectTableFunction`: `self.snowflake_plan.schema_query`
6. `SelectStatement`: 
```
        if self._schema_query:
            return self._schema_query
        if not self.projection:
            self._schema_query = self.from_.schema_query
            return self._schema_query
        self._schema_query = f"{analyzer_utils.SELECT}{self.projection_in_str}{analyzer_utils.FROM}({self.from_.schema_query})"
        return self._schema_query
```

for 1, 2, 3 schema_query is independent of subquery plans
for 4, 5 schema_query will be updated when we call `self._snowflake_plan = self._snowflake_plan.with_subqueries(subquery_plans)`

this only needs to be handled for `SelectStatement`